edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/device.rs

```diff
@@ -0,0 +1,385 @@
+//! This module contains the device client and device message types.
```

Comment:
maybe the better name for this module is `client.rs`?

Reply:
I just realized that the you have the same struct in module.rs. Which one is used?

Reply:
Oh, I see, it is device vs module clients. But a lot of logic is duplicated, anyways we can avoid that?

Reply:
Sorry I should have mentioned, this is Arnav SDK. I am only use the code as a client